pkg/registry/core/pod/strategy.go

```diff
@@ -89,9 +93,31 @@ func (podStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
+func isUpdatingUninitializedPod(old runtime.Object) (bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.Initializers) {
```

Comment:
The generic registry will delete the initiailizers if the feature gate is disabled, so the check here is just to be future-proof.